src/ui/component/dateInput/dateInput.tsx

```diff
@@ -0,0 +1,55 @@
+import { useMemo, type FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Icon } from '@/ui/component/icon/icon'
+import './dateInput.scss'
+import type { PartialDateTimeFormatOptions } from '@/util/date/date'
+import { formatDateToPattern } from '@/util/date/date'
+
+type DateInputProps = {
+  date?: Date
+}
+
+export const DateInput: FC<DateInputProps> = ({ date }) => {
+  const { i18n } = useTranslation()
+  const locale = i18n.language
+
+  const pattern = useMemo(
+    () => ({
+      day: 'DD',
+      month: 'MM',
+      year: 'YY'
+    }),
+    []
+  )
+
+  const dateFormatOptions: PartialDateTimeFormatOptions = useMemo(
+    () => ({
+      day: '2-digit',
+      month: '2-digit',
+      year: 'numeric'
+    }),
+    []
+  )
+
+  const dateFormatter = useMemo(
+    () => new Intl.DateTimeFormat(locale, dateFormatOptions),
+    [dateFormatOptions, locale]
+  )
+
+  const datePattern = useMemo(
```

Comment:
Just as a minor remark, I would have named it `datePlaceholder` or something like that because it's not really a pattern. 

The pattern is much more the model itself in which you want to map the date. 

And, moreover, it seems a bit confusing to have the `pattern` constant in line 16 (that is a date pattern) and here another pattern variable which is the return type of the `formatDateToPattern` function.

Reply:
yes it makes more sense, variables has been renamed. Thanks